RT-5.1: Singleton Interface

cprabha opened the pull request:
Hi,

1.) Mac address is not getting populated and hence added code to get it using gnmi.get
2.) Packets counters for larger MTU will be updated in /ethernet/counters , hence added check to verify them.

Thanks,
Prabha

ming2004 commented on feature/interface/singleton/ate_tests/singleton_test/singleton_test.go:
```diff
@@ -417,17 +430,24 @@ func (tc *testCase) testFlow(t *testing.T, packetSize uint16, ipHeader ondatra.H
-		t.Errorf("DUT received too few source packets: got %d, want >= %d", p1InDiff.unicast, outPkts)
-	}
 
 	if inPkts == 0 {
```
Where is the end curly brace for
if inPkts == 0 {
since you deleted line 428, 429, and a new if block starts from 439?

This will cause syntax error, right? can this be compiled? Does GH do such kind of sanity check?

cprabha replied:
Please check this block of code, it is passing and i  will attach passed log.

if inPkts == 0 {
		// The PacketLargerThanMTU cases do not expect to receive packets,
		// so this is not an error. Ethernet counters will be incremented
		// in case of Packets which larger than MTU
		t.Log("Flow did not receive any packet")
		if p1InEthDiff.unicast < outPkts {
			t.Errorf("DUT received too few source packets: got %d, want >= %d", p1InEthDiff.unicast, outPkts)
		}
	} else {
		if p1InDiff.unicast < outPkts {
			t.Errorf("DUT received too few source packets: got %d, want >= %d", p1InDiff.unicast, outPkts)
		}
		if avg := octets / inPkts; avg > uint64(tc.mtu) {
			t.Errorf("Flow destination packet size average got %d, want <= %d (MTU)", avg, tc.mtu)
		}
	}

cprabha commented:
[singleton_6.txt](https://github.com/openconfig/featureprofiles/files/10440667/singleton_6.txt)
Please check passed log.

Thanks,
Prabha

ming2004 commented on feature/interface/singleton/ate_tests/singleton_test/singleton_test.go:
```diff
@@ -417,17 +430,24 @@ func (tc *testCase) testFlow(t *testing.T, packetSize uint16, ipHeader ondatra.H
-	} else if avg := octets / inPkts; avg > uint64(tc.mtu) {
-		t.Errorf("Flow destination packet size average got %d, want <= %d (MTU)", avg, tc.mtu)
+		if p1InEthDiff.unicast < outPkts {
+			t.Errorf("DUT received too few source packets: got %d, want >= %d", p1InEthDiff.unicast, outPkts)
```
this message has no difference with line 444. Can you please make them different and more precise to differentiate the error cases?

cprabha replied:
updated please check.

cprabha commented:
[singleton_latest.txt](https://github.com/openconfig/featureprofiles/files/10470794/singleton_latest.txt)
Please find passed log.

ming2004 commented:
Seems there is some change when I'm verifying this PR. Can you please confirm if the PR need rebase?

cprabha commented:
> Seems there is some change when I'm verifying this PR. Can you please confirm if the PR need rebase?

@ming2004 I saw the changes and did rebase also removed certain parts of code as it is removed now. 
Currently we have added change  only in two places . To populate mac address and configure description for interface.

cprabha commented:
and it is rebased .
